bundles/framework/publisher2/instance.js

```diff
@@ -129,8 +123,140 @@ Oskari.clazz.define('Oskari.mapframework.bundle.publisher2.PublisherBundleInstan
+                    // Check that the data contains selected layers configuration
+                    var dataHasLayersConfig =
+                        data &&
+                        data.configuration &&
+                        data.configuration.mapfull &&
+                        data.configuration.mapfull.state &&
+                        data.configuration.mapfull.state.selectedLayers;
+
+                    if (dataHasLayersConfig) {
+                        var selectedLayers = data.configuration.mapfull.state.selectedLayers;
+                        var mapLayerService = me.sandbox.getService('Oskari.mapframework.service.MapLayerService');
+                        var timeoutInMillis = 1000; // one second;
+                        var maxTries = 30;
+                        var numTries = 0;
+
+                        // Interval checks if layers have been loaded.
+                        var intervalId = setInterval(function () {
+                            var layerIds = selectedLayers.map(function (l) { return l.id; });
+                            var layersFound = mapLayerService.hasLayers(layerIds);
+
+                            // try until we get the layers...
+                            if (layersFound) {
+                                if (infoDialog) {
+                                    infoDialog.close();
+                                }
+                                clearInterval(intervalId);
+                                me.setPublishMode(true, data);
+                                return;
+                            }
+                            numTries++;
+
+                            // ...or the max try count is reached
+                            if (numTries === maxTries) {
+                                clearInterval(intervalId);
+                                if (infoDialog) {
+                                    infoDialog.fadeout();
+                                }
+                                me._showOpeningPublisherErrorMsg();
+                            }
+                        }, timeoutInMillis);
+                    } else {
+                        if (infoDialog) {
+                            infoDialog.fadeout();
+                        }
+                        me._showOpeningPublisherErrorMsg();
+                    }
+                    // Perform layer check only once when opening the publisher on startup.
+                    layerCheckDone = true;
+                } else {
+                    me.setPublishMode(true, data);
+                }
+            };
+        },
+        /**
+         * @private @method _sendEditRequest
+         * Sends Publisher.PublishMapEditorRequest with given uuid.
+         *
+         * @param uuid {String} Uuid for the view to edit.
+         */
+        _sendEditRequest: function (uuid) {
+            var publishMapEditorRequestBuilder = this.sandbox.getRequestBuilder(
```

Comment:
Use Oskari.requestBuilder('Publisher.PublishMapEditorRequest') instead and you can assume the builder exists as it's part of the same publisher bundle.